### PR TITLE
Create Spelling, String for Enums and Error for Error-Enums Fixes #14

### LINKED ISSUE #14: Do correct String() or Error() Methods for enums
- Errors should have an Error() method
- Use Spelling() in String() if it exist
- Generate String() out of the Enum's items

```go
func (ck CommentKind) String() string {
    switch ck {
    case Comment_Null:
        return "Null"
    case Comment_Text:
        return "Text:"
    default:
             return fmt.Sprintf(“Unknown CommentKind value (%d)”, int(ck))
    }
}

func (c CursorKind) to_c() uint32 {
    return uint32(c)
}

func (c CursorKind) String() string {
    return c.Spelling()
}

/* for debug/testing */
func (c CursorKind) Spelling() string {
    cstr := cxstring{C.clang_getCursorKindSpelling(c.to_c())}
    defer cstr.Dispose()
    return cstr.String()
}

func (tle TypeLayoutError) Error() string {
    switch tle {
    case TLE_Invalid:
        return "TypeLayout=Invalid"

    case TLE_Incomplete:
        return "TypeLayout=Incomplete"

    case TLE_Dependent:
        return "TypeLayout=Dependent"

    case TLE_NotConstantSize:
        return "TypeLayout=NotConstantSize"

    case TLE_InvalidFieldName:
        return "TypeLayout=InvalidFieldName"
    }

    return fmt.Sprintf("TypeLayout=unknown (%d)", int(tle))
}
```

### PR DESCRIPTION
**Pull request by marxriedl**

### CODE REVIEW

File: cmd/go-clang-generate/enum.go [hunk 1]
```diff
@@ -122,3 +124,128 @@ func generateEnum(e *Enum) error {
...
+	}
+	if !hasString {
+		generateEnumMethod(e, templateGenerateEnumString)
+	}
+	if strings.HasSuffix(e.Name, "Error") && !hasError {
+		generateEnumMethod(e, templateGenerateEnumError)
+	}
+}
+
+func generateEnumSpelling(e *Enum, temp *template.Template) error {
```
Comment (zimmski) [inline cmd/go-clang-generate/enum.go]: Rename "temp" -> "tmpl"
Rename "generateEnumSpelling" -> "generateEnumSpellingMethod"

File: cmd/go-clang-generate/enum.go [hunk 2]
```diff
@@ -122,3 +124,128 @@ func generateEnum(e *Enum) error {
...
+		}
+	}
+
+	if !hasSpelling {
+		err := generateEnumSpelling(e, templateGenerateEnumSpelling)
+		if err != nil {
+			fmt.Println(err)
+		}
+	}
+	if !hasString {
```
Comment (zimmski) [inline cmd/go-clang-generate/enum.go]: This will always happe since hasString is not set

File: cmd/go-clang-generate/enum.go [hunk 3]
```diff
@@ -122,3 +124,128 @@ func generateEnum(e *Enum) error {
...
 	return ioutil.WriteFile(strings.ToLower(e.Name)+"_gen.go", b.Bytes(), 0600)
 }
+
+func generateEnumStringMethods(e *Enum) {
+	hasSpelling := false
+	hasString := false
+	hasError := false
+
+	for _, fStr := range e.Methods {
+		if strings.Contains(fStr, ") Spelling() ") {
```
Comment (zimmski) [inline cmd/go-clang-generate/enum.go]: All three ifs in this loop set the same variable -> put them in one condition

Comment (zimmski) [inline cmd/go-clang-generate/enum.go]: I guess you meant to use the other variables too?

File: cmd/go-clang-generate/main.go [hunk 4]
```diff
@@ -454,6 +454,7 @@ func main() {
 	}
 
 	for _, e := range enums {
+		generateEnumStringMethods(e)
```
Comment (zimmski) [inline cmd/go-clang-generate/main.go]: Do the call inside of generateEnum(...)

**Comment by zimmski** [comment]
Done. This can be easily done with the AST generation. But let's do that in another PR.

File: cmd/go-clang-generate/enum.go (hunk 1 above)
Comment (marxriedl) [inline cmd/go-clang-generate/enum.go]: fixed

File: cmd/go-clang-generate/enum.go (hunk 2 above)
Comment (marxriedl) [inline cmd/go-clang-generate/enum.go]: fixed

File: cmd/go-clang-generate/enum.go (hunk 3 above)
Comment (marxriedl) [inline cmd/go-clang-generate/enum.go]: yep

Comment (marxriedl) [inline cmd/go-clang-generate/enum.go]: fixed

File: cmd/go-clang-generate/main.go (hunk 4 above)
Comment (marxriedl) [inline cmd/go-clang-generate/main.go]: fixed

**Pull request merged**